src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java

```diff
@@ -0,0 +1,234 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.minecolonies.blockout.Log;
+import com.minecolonies.blockout.views.Window;
+import com.minecolonies.coremod.blocks.BlockHutDeliveryman;
+import com.minecolonies.coremod.client.gui.WindowWareHouseBuilding;
+import com.minecolonies.coremod.colony.Colony;
+import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.tileentities.TileEntityColonyBuilding;
+import com.minecolonies.coremod.tileentities.TileEntityWareHouse;
+import io.netty.buffer.ByteBuf;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Class of the warehouse building.
+ */
+public class BuildingWareHouse extends AbstractBuilding
+{
+    /**
+     * String describing the Warehouse.
+     */
+    private static final String WAREHOUSE = "WareHouse";
+
+    /**
+     * Tag to store the deliverymen.
+     */
+    private static final String TAG_DELIVERYMAN = "Deliveryman";
+
+    /**
+     * The list of deliverymen registered to this building.
+     */
+    private static final List<Vec3d> registeredDeliverymen = new ArrayList<>();
```

Comment:
What are these vec's pointing at? the buildings or the actual entities? I don't see why we can't use BlockPos if its buildings